### DISCUSSION TITLE
Rate limits when querying multiple conversation IDs

### DISCUSSION THREAD

**Discussion by anandwashere** [cjbarrie/academictwitteR#227] [Q&A] [answered]
I have a list of tweets, and I'm looking to pull all tweets that quote or share conversation IDs with my list. 
What I've attempted is a for loop that goes through my originial list and looks for quotes and replies.  See below:

`tweets_raw<-c()`

`build_query2 <- function(conversation_id) {
  glue::glue("conversation_id:{conversation_id} OR (url:{conversation_id} is:quote)", conversation_id = conversation_id)
}`

`for (i in 1:dim(tweets_text)[1]){
 ` 
`x <- get_all_tweets(
  build_query2(conversation_id = tweets_text[i,2]),
  start_tweets=tweets_text$created_at[i],
  end_tweets=paste(as.Date(tweets_text$created_at[1])+5,"T23:59:59Z",sep=""),
  n= Inf,
  context_annotations=F,bind_tweet=T,
  bearer_token=bearer_token)
tweets_raw<-rbind(tweets_raw,x)
print(i)
} `

However, I'm running into a rate limit for each step in the loop. Any idea why this would happen? I'm not hitting 300 tweets or whatever the rate limit is for academic track v2 full archive endpoint.  My original list of tweets is ~130K long, and most have no quotes or replies. As it stands, my solution is going to take very long to run. Any better solutions?

`query:  conversation_id:1312706526101020674 OR (url:1312706526101020674 is:quote) 
Total pages queried: 1 (tweets captured this page: 0).
This is the last page for conversation_id:1312706526101020674 OR (url:1312706526101020674 is:quote) : finishing collection.
[1] 1
query:  conversation_id:1127680734498557952 OR (url:1127680734498557952 is:quote) 
Rate limit reached. Rate limit will reset at 2021-08-23 10:23:50 
Sleeping for 780 seconds.`

**Comment by chainsawriot** [comment, marked as answer]
#195

**Reply by anandwashere** [reply]
That's perfect, thank you!
